### PR TITLE
feat(conf): logic group template

### PR DESCRIPTION
**Pull request by Roagert**

### CODE REVIEW

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+				"value": 2
+			},
+			{
+				"label": "Same as 1",
+				"value": 3
+			},
+			{
+				"label": "Same as 2",
+				"value": 4
+			},
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/templates/logic_group_template.json]: UIs probably show the label, not the value. So these labels aren't terribly helpful

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+				"value": 1
+			},
+			{
+				"label": "Color for ON state",
+				"value": 2
+			}
+		]
+	},
+	"dimmer_pushbutton": {
+		"$import": "../../templates/master_template.json#base_0-255_nounit",
```
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json:147]: Same for all other parameters where you impor this and then redefine half of it.

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+				"value": 0
+			},
+			{
+				"label": "Byte 1: enabled – a single activation will send commands to devices in association group 4. devices will receive commands with the values set in byte 2 and 3 (default)",
+				"value": 1
+			},
+			{
+				"label": "Byte 2: when single pressing the pushbutton for on, a multilevel switch set with this value will be send to devices in association group 4. (default)",
+				"value": 255
+			}
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/templates/logic_group_template.json]: This smells like partial parameters, but these options make no sense at all.

File: packages/config/config/devices/0x0234/zba7140.json [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+		"18[0xff000000]": {
+			"$import": "templates/logic_group_template.json#multilevel_duration_function",
+			"label": "Pushbutton 4 Double-Activation Duration Value"
+		}
+	},
+	"metadata": {
+		"inclusion": "Remove white pushbutton cover and press on the center switch with a non-conductive object. The LEDs will now start blinking on pushbutton 1 (upper left pushbutton)",
+		"exclusion": "Remove white pushbutton cover and press on the center switch with a non-conductive object. The LEDs will now start blinking on pushbutton 1 (upper left pushbutton)",
+		"reset": "Remove white pushbutton cover and long-press the center switch for 10 seconds with a non-conductive object. Please use this procedure only when the network primary controller is missing or otherwise inoperable",
+		"manual": ""
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zba7140.json]: Don't add empty keys please

File: packages/config/config/devices/0x0234/zdb5100.json [hunk 5]
```diff
@@ -4,824 +4,494 @@
...
 	],
 	"firmwareVersion": {
 		"min": "0.0",
 		"max": "255.255"
 	},
-	"supportsZWavePlus": true,
+	"associations": {
+		"1": {
+			"label": "Lifeline",
+			"description": "Sends Device Reset notifications, and Central Scene notifications.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zdb5100.json]:
```suggestion
```

File: packages/config/config/devices/0x0234/zdb5100.json [hunk 6]
```diff
@@ -4,824 +4,494 @@
...
+			"maxNodes": 5,
+			"isLifeline": true
+		},
+		"2": {
+			"label": "Basic Report, Pushbutton 1",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zdb5100.json:26]: Please make the association labels consistent between the files

File: packages/config/config/devices/0x0234/zdb5100.json [hunk 7]
```diff
@@ -4,824 +4,494 @@
...
-			"description": "Time that a button must be activated before it is accepted as “held-down\".",
-			"valueSize": 1,
-			"minValue": 1,
-			"maxValue": 255,
-			"defaultValue": 50,
+			"unit": "0.01 seconds",
+			"defaultValue": 50
+		},
+		"14[0xff]": {
+			"$import": "../templates/master_template.json#base_0-255_nounit",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zdb5100.json]: See comments above

File: packages/config/config/devices/0x0234/zdb5100.json [hunk 8]
```diff
@@ -4,824 +4,494 @@
...
+			"defaultValue": 127,
 			"allowManualEntry": true
 		},
+		"15": {
+			"label": "Associations groups",
+			"$import": "../templates/master_template.json#base_0-255_nounit",
+			"valueSize": 2,
+			"unsigned": true,
+			"maxValue": 65535,
+			"defaultValue": 8191,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zdb5100.json]: That smells like partial params

File: packages/config/config/devices/0x0234/zdb5100.json [hunk 9]
```diff
@@ -4,824 +4,494 @@
...
+			"allowManualEntry": true
+		},
+		"16": {
+			"$import": "templates/logic_group_template.json#pushbutton_function_zdb5100",
+			"label": "Pushbutton 1 functionality"
+		},
+		"17": {
+			"$import": "../templates/master_template.json#base_0-255_nounit",
+			"label": "Timer value for pushbutton 1",
+			"unit": "1.0 seconds",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zdb5100.json]:
```suggestion
			"unit": "seconds",
```

File: packages/config/config/devices/0x0234/zdb5100.json [hunk 10]
```diff
@@ -4,824 +4,494 @@
...
+		"22[0xff0000]": {
+			"$import": "templates/logic_group_template.json#led_indicator",
+			"label": "Pushbutton 1 LED indicator ON Blue",
+			"valueSize": 4,
+			"defaultValue": 127
+		},
+		"22[0xff000000]": {
+			"$import": "../templates/master_template.json#base_0-255_nounit",
+			"label": "LED Time for pushbutton 1 blinking",
+			"unit": "x0.1 sec",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zdb5100.json]:
```suggestion
			"unit": "0.1 seconds",
```
Please check the style guide so your units are consistent

File: packages/config/config/devices/0x0234/zdb5100.json [hunk 11]
```diff
@@ -4,824 +4,494 @@
...
+			"valueSize": 4,
+			"defaultValue": 0,
 			"allowManualEntry": true
 		},
-		"23": {
-			"label": "OFF indication RGB-saturation for pushbutton 1",
+		"23[0xff]": {
+			"$import": "templates/logic_group_template.json#led_indicator",
+			"label": "Pushbutton 1 LED indicator OFF Red",
+			"valueSize": 4,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zdb5100.json]: Shouldn't this be in the template?

File: packages/config/config/devices/0x0234/zif5028.json [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+		"24": {
+			"$import": "templates/logic_group_template.json#output_function_relay",
+			"label": "Function setup of input 6"
+		}
+	},
+	"metadata": {
+		"inclusion": "Push the button in the hole near the 'INCLUSION' text with a non-conductive object. The LED will now start blinking",
+		"exclusion": "Push the button in the hole near the 'INCLUSION' text with a non-conductive object. The LED will now start blinking",
+		"reset": "Push the button in the hole near the 'INCLUSION' text with a non-conductive object for 10 seconds. Please use this procedure only when the network primary controller is missing or otherwise inoperable",
+		"manual": ""
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0234/zif5028.json]:
```suggestion
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]
This is gonna need a 2nd pass - mainly for consistency and conformance with the style guide.

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 13]
```diff
@@ -0,0 +1,474 @@
...
+			},
+			{
+				"label": "Current level of the battery",
+				"value": 1
+			},
+			{
+				"label": "LED colour is set by configuration parameter 5.",
+				"value": 2
+			}
+		]
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: What is your suggestion here then? Just have a base_nounit_0-255 instead and add options ?

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 14]
```diff
@@ -0,0 +1,474 @@
...
+		"description": "Lower switch value",
+		"valueSize": 4
+	},
+	"multilevel_duration_function": {
+		"$import": "../../templates/master_template.json#base_0-255_nounit",
+		"description": "Duration",
+		"valueSize": 4
+	},
+	"pushbutton_functionality": {
+		"$import": "../../templates/master_template.json#base_0-255_nounit",
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: i can change it to this: led_indicator_four_options instead but its a button not an led ?

File: packages/config/config/devices/0x0234/zba7140.json (hunk 4 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zba7140.json]: What link should i add for the manual if the manual is not located in the ZWA database ?

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: Probably no base at all, just define the parameter here. Maybe @blhoward2 has a preference after working on the Aeotec files.

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 14 above)
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: See comment above, probably just don't import at all.

File: packages/config/config/devices/0x0234/zba7140.json (hunk 4 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x0234/zba7140.json]: Nothing, just don't add that line.

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 14 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: i understand. Thank you

File: packages/config/config/devices/0x0234/zba7140.json (hunk 4 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zba7140.json]: Understood, and removed for future push

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 15]
```diff
@@ -0,0 +1,474 @@
...
+				"value": 64
+			},
+			{
+				"label": "Messages in association group 9 are sent as secure",
+				"value": 128
+			},
+			{
+				"label": "All messages in all groups are sent as secure",
+				"value": 255
+			}
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: I understand that this is a mess, i am trying to do something that looks like this now:

Template:
	"association_param_secure": {
		"label": "Associations groups",
		"valueSize": 2,
		"minValue": 0,
		"maxValue": 255,
		"defaultValue": 255,
		"readOnly": false,
		"writeOnly": false,
		"allowManualEntry": true
	},


Device file:
		"6[0x00]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"value": 0
		},
		"6[0x01]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 2 are sent as secure",
			"value": 1
		},
		"6[0x02]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 3 are sent as secure",
			"value": 2
		},
		"6[0x04]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 4 are sent as secure",
			"value": 4
		},
		"6[0x08]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 5 are sent as secure",
			"value": 8
		},
		"6[0x16]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 6 are sent as secure",
			"value": 16
		},
		"6[0x32]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 7 are sent as secure",
			"value": 32
		},
		"6[0x64]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 8 are sent as secure",
			"value": 64
		},
		"6[0x80]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group 9 are sent as secure",
			"value": 128
		},
		"6[0xff]": {
			"$import": "templates/logic_group_template.json#association_param_secure",
			"label": "Messages in association group all are sent as secure",
			"value": 255
		},

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 16]
```diff
@@ -0,0 +1,474 @@
...
+			},
+			{
+				"label": "Push Button 3",
+				"value": 4
+			},
+			{
+				"label": "Push Button 4",
+				"value": 8
+			}
+		]
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: this should be a partial parameter with enable and disable like so ? 
			"1[0x00]":{
                                "$import": "../templates/master_template.json#base_enable_disable",
				"label": "Disabled",
				"defaultValue": 0
			},
                       "1[0x02]":{
                                "$import": "../templates/master_template.json#base_enable_disable",
				"label": "Pushbutton 1",
				"defaultValue": 1
			},

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: Alright, i have used no base instead now and ofcourse removed the n in the options that is my bad. Good catch

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json:147]: color_command_pushbnutton doesnt have an import ?

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 15 above)
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: That doesn't work that way. You'll need options for association groups 2-9. 0 just means none of them are set, 255 means all of them are set - those two options need no extra parameter.

Each one of those partial parameters is just a single bit, so values 0/1 which are probably best represented as Enabled/Disabled options.

The base parameter you're importing also needs a very specific setup. Probably best if you read 
https://zwave-js.github.io/node-zwave-js/#/config-files/file-format?id=partial-parameters
again plus the additional explanation under
https://zwave-js.github.io/node-zwave-js/#/config-files/partial-parameters

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json:147]: What do you mean?

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 16 above)
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: You don't copy the values 1:1 into partial parameters. Each one that can be enabled (Pushbuttons 1..4) needs to be represented as a partial. See my comment above.

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 15 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: right ofcourse that makes sense. Its still a little confusing at times. i will make sure to change that

File: packages/config/config/devices/0x0234/zdb5100.json (hunk 11 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zdb5100.json]: resolved in the lates update

File: packages/config/config/devices/0x0234/zdb5100.json (hunk 10 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zdb5100.json]: resolved in the latest update

File: packages/config/config/devices/0x0234/zdb5100.json (hunk 9 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zdb5100.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: .gitignore [hunk 17]
```diff
@@ -1,34 +0,0 @@
-node_modules/
```
[the remark this answers is not in the document]
Comment (Roagert) [reply .gitignore]: my bad. incorrect file upload

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 18]
```diff
@@ -0,0 +1,474 @@
...
+				"value": 1
+			},
+			{
+				"label": "LED colour is set by configuration parameter 5.",
+				"value": 2
+			}
+		]
+	},
+	"association_param_secure": {
+		"label": "Associations groups, transmission when included secure.",
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: label and partial parameter updated

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 15 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 16 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 14 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 19]
```diff
@@ -0,0 +1,474 @@
...
+		]
+	},
+	"multilevel_enable": {
+		"$import": "../../templates/master_template.json#base_enable_disable",
+		"valueSize": 4,
+		"defaultValue": 1
+	},
+	"multilevel_upper_function": {
+		"$import": "../../templates/master_template.json#base_0-99_nounit",
+		"description": "Upper switch value",
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: correct good catch

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 20]
```diff
@@ -0,0 +1,474 @@
...
+		"valueSize": 4,
+		"defaultValue": 255
+	},
+	"multilevel_lower_function": {
+		"$import": "../../templates/master_template.json#base_0-99_nounit",
+		"description": "Lower switch value",
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: correct good catch

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 21]
```diff
@@ -0,0 +1,474 @@
...
+		"valueSize": 4
+	},
+	"multilevel_duration_function": {
+		"$import": "../../templates/master_template.json#base_0-255_nounit",
+		"description": "Duration",
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: correct good catch

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json:147]: What is the highlight here ? that i import the base unit but changes the maxValue ?

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 22]
```diff
@@ -0,0 +1,474 @@
...
+				"value": 0
+			},
+			{
+				"label": "Duration in seconds",
+				"value": 127
+			},
+			{
+				"label": "Duration in minutes (minus 127) from 1 – 128 minutes, where 128 is 1 minute",
+				"value": 255
+			}
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 23]
```diff
@@ -0,0 +1,474 @@
...
+				"value": 0
+			},
+			{
+				"label": "Time in seconds",
+				"value": 127
+			},
+			{
+				"label": "Time in minutes (minus 127) from 1 – 128 minutes, where 128 is 1 minute",
+				"value": 255
+			}
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 24]
```diff
@@ -0,0 +1,474 @@
...
+		]
+	},
+	"dimmer_mode": {
+		"$import": "../../templates/master_template.json#base_0-255_nounit",
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 25]
```diff
@@ -0,0 +1,474 @@
...
+	"dimmer_max": {
+		"$import": "../../templates/master_template.json#base_0-99_nounit",
+		"label": "Dimmer maximum level",
+		"unit": "%",
+		"defaultValue": 99
+	},
+	"led_indicator": {
+		"$import": "../../templates/master_template.json#base_0-255_nounit",
+		"valueSize": 4,
+		"unsigned": true,
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: Why what ?

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 1 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: resolved

File: packages/config/config/devices/0x0234/zba7140.json (hunk 4 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zba7140.json]: resolved

File: packages/config/config/devices/0x0234/zdb5100.json (hunk 8 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zdb5100.json]: resolved

File: packages/config/config/devices/0x0234/templates/logic_group_template.json [hunk 26]
```diff
@@ -0,0 +1,474 @@
+{
+	//Generic for Multiple Devices in Logic Group
+	"enable_central_scene": {
+		"$import": "../../templates/master_template.json#base_enable_disable",
+		"label": "Central Scene notifications",
```
[the remark this answers is not in the document]
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: Should be resolved.

File: packages/config/config/devices/0x0234/templates/logic_group_template.json (hunk 3 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/templates/logic_group_template.json]: should be resolved in this update that is inc.

File: packages/config/config/devices/0x0234/zdb5100.json (hunk 6 above)
Comment (Roagert) [reply packages/config/config/devices/0x0234/zdb5100.json:26]: resolved

**Comment by Roagert** [comment]
Please review the files, the template has been updated including checking all of your comments has been resolved.
More devices has been added that were missing to the logic group manufacturer

**Comment by Roagert** [comment]
@AlCalzone i have updated the files to accomodate your review including adding missing devices and Update the template for those devices

**Comment by AlCalzone** [comment]
@blhoward2 can you take a look please?

**Comment by Roagert** [comment]
@blhoward2 do you have any input ?

**Comment by blhoward2** [comment]
I'll look when I'm able. I haven't had a chance. Work stuff.

**Pull request closed without merging**